open-craft/learning-paths-plugin
# [BB-8488] Link course enrollments to learning pathways

## Description
This PR links course enrollments to learning pathways for more granular control over individual course enrollments. It extends the admin feature to bulk enroll users into the courses in the pathway.

## Testing Instructions
1. Set up devstack
2. Clone this repository to `<devstack root>/src`
3. Start lms: `$ make dev.up.lms`
4. Enter lms shell: `$ make dev.shell.lms`
5. Install this plugin: `# pip install -e /edx/src/learning-paths-plugin/`
6. Run migrations: `# ./manage.py lms migrate learning_paths`
7. Restart lms: `$ make dev.stop.lms dev.up.lms`
8. Login as `verified` user and check dashboard for enrolled courses.
9. As an admin user, create a learning path at http://localhost:18000/admin/learning_paths/learningpath/ with a few courses in the learning steps section.
10. In the "Bulk enroll users" field, add `verified` username and save the learning path.
11. Login again as `verified` user and verify that the user has been enrolled in all the courses listed in the learning steps section.
